http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -97,6 +97,21 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         s"Begin app creation for app ${params.appName.value} in cloud context ${params.cloudContext.asString}"
       )
 
+      referenceDatabaseNames = app.appType.databases.collect { case ReferenceDatabase(name) => name }.toSet
+
+      // build WSM resource client
+      wsmResourceApi <- buildWsmResourceApiClient
+      wsmResourceDatabases <- F.delay(
```

Comment:
As an optimization, could we only do this is the app _has_ any reference databases? 
```
referenceDatabaseNames = app.appType.databases.collect { case ReferenceDatabase(name) => name }.toSet
referenceDatabases <- if (referenceDatabaseNames.nonEmpty) {
  retrieveWsmReferenceDatabases(…)
} else F.unit
```